kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/BaseContentService.java

```diff
@@ -7450,6 +7450,28 @@ public void handleAccess(HttpServletRequest req, HttpServletResponse res, Refere
 				String refId = ref.getId();
 				if (refId == null) refId = "";
 
+				// We allow access to dropbox files because they are linked to the user, to avoid this we should restrict to /group-user/.
+				// We don't allow access to contents or collections if the site is unpublished and the user cannot visit unpublished sites.
+				// We don't allow access to attachments if the site is unpublished and the user cannot visit unpublished sites.
+				// We allow access to public files or collections, independently of the site permissions.
+				if (ref.getId().startsWith("/group/") || ref.getId().startsWith("/attachment/")) {
+					try {
+						//When a file or collection is public, there is a realm associated to the file or the collection.
+						m_authzGroupService.getAuthzGroup(ref.getReference());
+					} catch (Exception ex) {
+						try {
+							//The public file realm doesn't exist, check the permissions.
+							String siteId = ref.getId().split("/")[2];
+							Site site = m_siteService.getSite(siteId);
+							if (!site.isPublished() && !m_securityService.unlock(sessionManager.getCurrentSessionUserId(), m_siteService.SITE_VISIT_UNPUBLISHED, siteId)) {
```

Comment:
Need a couple of checks here, for nulls.

Reply:
Don't think so, there is a try catch statement, if anything happens it restricts the access to the file. What do you suggest?

Reply:
You're throwing an entitypermissionexception when it may just be that there's no site for that id, or the original ref may not be of the right format. I'd just make that code a bit more defensive, that's all. Check the split array in advance, do a bit of null checking. That way the log messages we get will make more sense.